crates/sparrow-runtime/src/execute/output.rs

```diff
@@ -94,38 +124,39 @@ pub(super) fn write(
             progress_updates_tx,
             batches,
         )
-        .change_context(Error::WritingToDestination {
-            dest_name: "object_store".to_owned(),
-        })
+        .change_context(Error::WritingToDestination("object_store"))
         .boxed()),
-        Destination::Redis(redis) => {
-            Ok(
-                redis::write(redis, sink_schema, progress_updates_tx, batches)
-                    .change_context(Error::WritingToDestination {
-                        dest_name: "redis".to_owned(),
-                    })
-                    .boxed(),
-            )
-        }
-        #[cfg(not(feature = "pulsar"))]
-        Destination::Pulsar(_) => {
-            error_stack::bail!(Error::FeatureNotEnabled {
-                feature: "pulsar".to_owned()
-            })
+        Destination::Channel(channel) => {
+            Ok(write_to_channel(batches, channel, progress_updates_tx).boxed())
         }
         #[cfg(feature = "pulsar")]
         Destination::Pulsar(pulsar) => {
             Ok(
                 pulsar::write(pulsar, sink_schema, progress_updates_tx, batches)
-                    .change_context(Error::WritingToDestination {
-                        dest_name: "pulsar".to_owned(),
-                    })
+                    .change_context(Error::WritingToDestination("pulsar"))
                     .boxed(),
             )
         }
     }
 }
 
+async fn write_to_channel(
+    mut batches: BoxStream<'static, RecordBatch>,
+    channel: tokio::sync::mpsc::Sender<RecordBatch>,
+    _progress_updates_tx: tokio::sync::mpsc::Sender<ProgressUpdate>,
+) -> error_stack::Result<(), Error> {
+    while let Some(next) = batches.next().await {
+        channel
+            .send(next)
+            .await
+            .map_err(|_e| error_stack::report!(Error::WritingToDestination("channel")))?;
+
+        // progress_updates_tx.send(ProgressUpdate::Output { num_rows })
```

Comment:
I believe this should happen here?

Reply:
It should -- I'm leaving it commented out because I'm not 100% sure it will work (eg., it could fill the progress channel if nobody listens). Will revisit once we have more hooked up.